File: app/src/main/java/com/lence/penguinsvskillerwhales/view/UpdatePresenter.java
package com.lence.penguinsvskillerwhales.view;

import android.graphics.Point;

import com.lence.penguinsvskillerwhales.model.Orca;
import com.lence.penguinsvskillerwhales.model.Organism;

public interface UpdatePresenter {
    void update(KotlinField list);
    void move(Point thisPoint);
    void breeding(Point thisPoint, Organism organism);
    void eat(Point thisPoint, Orca orca);
    void death(Point thisPoint);
}

File: app/src/main/java/com/lence/penguinsvskillerwhales/view/UpdatePresenter.kt
package com.lence.penguinsvskillerwhales.view

import android.graphics.Point

import com.lence.penguinsvskillerwhales.model.Orca
import com.lence.penguinsvskillerwhales.model.Organism

interface UpdatePresenter {
    fun update(list: Field)
    fun move(thisPoint: Point)
    fun breeding(thisPoint: Point, organism: Organism)
    fun eat(thisPoint: Point, orca: Orca)
    fun death(thisPoint: Point)
}
